packages/cli/src/commands/database/seed.ts

```diff
@@ -0,0 +1,135 @@
+import { readdir, readFile } from 'fs/promises';
+import path from 'path';
+
+import { seeds } from '@logto/schemas';
+import {
+  createPool,
+  DatabasePool,
+  DatabaseTransactionConnection,
+  parseDsn,
+  sql,
+  stringifyDsn,
+} from 'slonik';
+import { raw } from 'slonik-sql-tag-raw';
+import { CommandModule } from 'yargs';
+import { z } from 'zod';
+
+import { createPoolFromConfig, getDatabaseUrlFromConfig, insertInto } from '../../database';
+import { buildApplicationSecret, log } from '../../utilities';
+
+/**
+ * Create a database pool with the database URL in config.
+ * If the given database does not exists, it will try to create a new database by connecting to the maintenance database `postgres`.
+ *
+ * @returns A new database pool with the database URL in config.
+ */
+const createDatabasePool = async () => {
+  try {
+    return await createPoolFromConfig();
+  } catch (error: unknown) {
+    const result = z.object({ code: z.string() }).safeParse(error);
+
+    // Database does not exist, try to create one
+    // https://www.postgresql.org/docs/14/errcodes-appendix.html
+    if (!(result.success && result.data.code === '3D000')) {
+      log.error(error);
+    }
+
+    const databaseUrl = await getDatabaseUrlFromConfig();
+    const dsn = parseDsn(databaseUrl);
+    // It's ok to fall back to '?' since:
+    // - Database name is required to connect in the previous pool
+    // - It will throw error when creating database using '?'
+    const databaseName = dsn.databaseName ?? '?';
+    const maintenancePool = await createPool(stringifyDsn({ ...dsn, databaseName: 'postgres' }));
```

Comment:
Should we allow custom database name, instead of hard-coded `postgres`?

Reply:
it's the maintenance database name, in most cases it'll be `postgres`. customization is good, should we implement until ask?

Reply:
ok, I'm good with it